addons/sale_timesheet/models/project_task.py

```diff
@@ -57,18 +58,31 @@ def _compute_remaining_hours_so(self):
     def _search_remaining_hours_so(self, operator, value):
         return [('sale_line_id.remaining_hours', operator, value)]
 
+    def _compute_last_sol_of_customer(self):
+        sol_per_domain = dict()
+        for task in self:
+            domain = tuple(task._get_last_sol_of_customer_domain())
+            if not domain:
+                task.last_sol_of_customer = False
+                continue
+            sol = sol_per_domain.get(domain)
+            if not sol:
+                sol = self.env['sale.order.line'].search(domain, limit=1)
+                sol_per_domain[domain] = sol
+            task.last_sol_of_customer = sol
```

Comment:
If search don't find a sol for a particular domain, you will search it again for no reason.
```suggestion
            if domain not in sol_per_domain:
                sol_per_domain[domain] = self.env['sale.order.line'].search(domain, limit=1)
            task.last_sol_of_customer = sol_per_domain[domain]
```

Reply:
Indeed, nice catch! :smile: